### ISSUE TITLE
matched_queries is empty when filtered query with nested filter is missing must_not clause

### ISSUE THREAD

**Issue by ehclark** [elastic/elasticsearch#13088]
The matched_queries item in the response is missing when using a filtered query with nested filter that does not have a must_not clause.

ES version 1.5.2

Steps to reproduce:

Create index and mapping
```
PUT /test_index
 
PUT /test_index/doc_type/_mapping
{
  "doc_type": {
    "dynamic": "strict",
    "properties": {
      "nestedDoc": {
        "type": "nested",
        "dynamic": "strict",
        "properties": {
          "field1": {
            "type": "string"
          }
        }
      }
    }
  }
}
```
Add a document
```
PUT /test_index/doc_type/1
{
  "nestedDoc": [
    {
      "field1": "a"
    },
    {
      "field1": "b"
    }
  ]
}
```
Nested filter query with only a must clause
```
GET /test_index/doc_type/_search
{
  "query" : {
    "filtered" : {
      "query" : {
        "match_all" : { }
      },
      "filter" : {
        "bool" : {
          "should" : {
            "bool" : {
              "must" : {
                "nested" : {
                  "filter" : {
                    "bool" : {
                      "must" : {
                        "term" : {
                          "field1" : "b"
                        }
                      }
                    }
                  },
                  "path" : "nestedDoc"
                }
              },
              "_name" : "QueryName"
            }
          }
        }
      }
    }
  }
}
```
Returns result - no matched_queries item as expected
```
{
   "took": 1,
   "timed_out": false,
   "_shards": {
      "total": 5,
      "successful": 5,
      "failed": 0
   },
   "hits": {
      "total": 1,
      "max_score": 1,
      "hits": [
         {
            "_index": "test_index",
            "_type": "doc_type",
            "_id": "1",
            "_score": 1,
            "_source": {
               "nestedDoc": [
                  {
                     "field1": "a"
                  },
                  {
                     "field1": "b"
                  }
               ]
            }
         }
      ]
   }
}
```
Nested filter query with an additional must_not clause that does nothing
```
GET /test_index/doc_type/_search
{
  "query" : {
    "filtered" : {
      "query" : {
        "match_all" : { }
      },
      "filter" : {
        "bool" : {
          "should" : {
            "bool" : {
              "must" : {
                "nested" : {
                  "filter" : {
                    "bool" : {
                      "must" : {
                        "term" : {
                          "field1" : "b"
                        }
                      }
                    }
                  },
                  "path" : "nestedDoc"
                }
              },
              "must_not" : {
                "exists" : { "field" : "_does_not_exist_" }
              },
              "_name" : "QueryName"
            }
          }
        }
      }
    }
  }
}
```
Returns result with expected matched_queries item
```
{
   "took": 1,
   "timed_out": false,
   "_shards": {
      "total": 5,
      "successful": 5,
      "failed": 0
   },
   "hits": {
      "total": 1,
      "max_score": 1,
      "hits": [
         {
            "_index": "test_index",
            "_type": "doc_type",
            "_id": "1",
            "_score": 1,
            "_source": {
               "nestedDoc": [
                  {
                     "field1": "a"
                  },
                  {
                     "field1": "b"
                  }
               ]
            },
            "matched_queries": [
               "QueryName"
            ]
         }
      ]
   }
}
```

**Comment by martijnvg** [comment]
@ehclark some bugs were fixed some time ago (#11880) with matched queries and inner docs and this particular issue is fixed in the next 2.0 beta release

One note, the field used in the `term` query is incorrect and should be `nestedDoc.field1` instead of `field1`.

**Comment by clintongormley** [comment]
Closing as duplicate of #11880
